manual/source/Infrastructure/Contribute/GitWorkflow.rst

```diff
@@ -34,59 +34,63 @@ Execute the following command to get the last sources:
-The most frequent development use cases are documented below.
+Almost all development occurs in develop.
+
+The master branch is only updated before a new release, either by merging from develop which will create a new minor release [2.x+1.0] or through an emergency fix which will result in a patch level release [2.x.y+1].
+Thus, the master branch is always stable and usually identical to the latest release, and the develop branch contains the latest development but might occasionally have breaking changes. 
 
 .. __: https://www.atlassian.com
 
-Develop a feature in a module or an app
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Develop a feature or fix a bug
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 Follow the `steps <https://www.atlassian.com/git/workflows#workflow-gitflow>`_ in “Mary and John begin new features” and “Mary finishes her feature”.
 
-* Create a new feature `branch <https://www.atlassian.com/git/tutorial/git-branches#branch>`_ based on `develop <https://github.com/seqan/seqan/tree/develop>`_.
+* Create a new `branch <https://www.atlassian.com/git/tutorial/git-branches#branch>`_ based on `develop <https://github.com/seqan/seqan/tree/develop>`_.
 * Perform your changes and `commit <https://www.atlassian.com/git/tutorial/git-basics#commit>`_ them onto your feature branch.
+* Keep your commit history lean (see below) and `write proper commit messages <infra-contribute-git-commits>`_.
 * When the development is complete, push the feature branch to your repository on GithHub.
+* Make sure that you have `signed the Contributor License Agreement <https://www.clahub.com/agreements/seqan/seqan>`_
 * `Create a GitHub pull request <https://github.com/seqan/seqan/compare/develop>`_ to `develop <https://github.com/seqan/seqan/tree/develop>`_.
-* Delete your feature branch once it has been merged.
-
-Fix an existing bug in a module or app
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+* Delete your branch once it has been merged.
 
-Follow the `steps <https://www.atlassian.com/git/workflows#workflow-gitflow>`_ in “End-user discovers a bug”.
+What to include in a commit
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-* Create a new hotfix `branch <https://www.atlassian.com/git/tutorial/git-branches#branch>`_ based on `master <https://github.com/seqan/seqan/tree/master>`_.
-* Perform your changes and `commit <https://www.atlassian.com/git/tutorial/git-basics#commit>`_ them onto your hotfix branch.
-* When the fix is ready, push your hotfix branch to repository on GitHub. Then:
-    #. `Create a GitHub pull request`__ to `master <https://github.com/seqan/seqan/tree/master>`_.
-    #. `Create a GitHub pull request`__ to `develop <https://github.com/seqan/seqan/tree/develop>`_.
-    #. The pull requests should contain only the commits from your hotfix branch.
-* Delete your hotfix branch once it has been merged through the pull request.
+The git history of your contribution should be concise and lean. Please follow the following hints:
```

Comment:
Still concise implies lean then already, so it is kind of superfluous, or is this some phrase?